packages/core/src/react/hooks/useFetchSearchNative.ts

```diff
@@ -0,0 +1,86 @@
+import { useFetch } from './useFetch';
+
+import type { FetchHookOptions, HookResponse } from './types';
+import {
+	FetchResponse,
+	getPostAuthor,
+	getPostTerms,
+	PageInfo,
+	PostSearchEntity,
+	TermSearchEntity,
+	SearchParams,
+	QueriedObject,
+	SearchNativeFetchStrategy,
+} from '../../data';
+import { getWPUrl } from '../../utils';
+import { makeErrorCatchProxy } from './util';
+
+export interface useSearchNativeResponse<
+	T extends PostSearchEntity | TermSearchEntity = PostSearchEntity | TermSearchEntity,
+> extends HookResponse {
+	data?: { posts: T[]; pageInfo: PageInfo; queriedObject: QueriedObject };
+}
+
+/**
+ * The useFetchSearchNative hook. Returns a collection of search entities retrieved through the WP native search endpoint
+ *
+ * See {@link useSearchNative} for usage instructions.
+ *
+ * @param params The list of params to pass to the fetch strategy. It overrides the ones in the URL.
+ * @param options The options to pass to the swr hook.
+ * @param path The path of the url to get url params from.
+ *
+ * @category Data Fetching Hooks
+ */
+export function useFetchSearchNative<
+	T extends PostSearchEntity | TermSearchEntity = PostSearchEntity | TermSearchEntity,
+	P extends SearchParams = SearchParams,
+>(
+	params: P | {} = {},
+	options: FetchHookOptions<FetchResponse<T[]>> = {},
+	path = '',
+): useSearchNativeResponse<T> {
+	const { data, error, isMainQuery } = useFetch<T[], P>(
+		params,
+		useFetchSearchNative.fetcher<T, P>(),
+		options,
+		path,
+	);
+
+	if (error || !data) {
+		const fakeData = {
+			posts: makeErrorCatchProxy<T[]>('posts'),
+			pageInfo: makeErrorCatchProxy<PageInfo>('pageInfo'),
+			queriedObject: makeErrorCatchProxy<QueriedObject>('queriedObject'),
+		};
+		return { error, loading: !data, data: fakeData, isMainQuery };
+	}
+
+	const { result, pageInfo, queriedObject } = data;
+
+	const posts = result.map((post) => {
```

Comment:
I think we'd want to name `posts` something else since it can be used to also search terms.

Alternativelly we keep `posts` if searching for post and `terms` when searching for terms.

Reply:
changed to searchResults as we also have post-format type